### DISCUSSION TITLE
Keeping Big Rat online forever

### DISCUSSION THREAD

**Discussion by dispherical** [bigratmonster/bigrat.monster#13] [Show and tell]
I feel like bigrat.monster is important. So, I took the power of IPFS and Git to help keep big rat online forever. IPFS works best if multiple people are hosting the site, so here is a command on how to add it

You will need to install the [ipfs cli tool](https://docs.ipfs.io/how-to/command-line-quick-start) for this. You can also use [jsipfs](https://www.npmjs.com/package/ipfs) (`npm install ipfs -g`), but you will have to change the `ipfs` commands to `jsipfs`. You will also need [git-scm*](https://git-scm.com/) to run it.

In one terminal, run `ipfs daemon`, in the other, run the following:

```bash
git clone https://github.com/bigratmonster/bigrat.monster.git
cd bigrat.monster
ipfs add -r .
```

You now have uploaded a copy of bigrat.monster to IPFS! Now, keeping bigrat.monster online requires keeping the daemon running, so you should use a system file or PM2 or something similar to keep big rat online.

~~Edit: [Big rat monster is now on IPFS!](https://gateway.ipfs.io/ipns/bigratmonster.aboutdavid.me) Due to me using IPNS, I've modified the tutorial to make it a lot easier to follow.~~
Edit 2: Looking at this, changes are made a lot, so it's best to clone the repository and add it that way

*git-scm is a fancy way of saying git, the version control system that powers github.

**Comment by bigratenthusiast** [comment]
Is this safe?

**Reply by dispherical** [reply]
Yes. You can learn more about IPFS [here](https://ipfs.io/)

**Comment by dispherical** [comment]
Whoever controls the domain, they should look into DNSlink, adding a TXT record to point the domain to the IPFS hash. I'll look into setting up a automated system to get hashes for bigrat.monster

**Comment by SoNothingMC** [comment]
Thanks David.

**Comment by smokeytube** [comment]
![image](https://user-images.githubusercontent.com/67290867/129459982-921aeb99-7d4b-492e-b7b5-9f8e1fa63ecd.png)
